src/rtt.cpp

```diff
@@ -0,0 +1,590 @@
+
+        if (initProbeStatus != SUCCESS)
+        {
+            highLevelFunctions.dll_close();
+            releaseHighLevel();
+            baton->lowlevelError = initProbeStatus;
+            return RTTCouldNotGetDeviceInformation;
+        }
+
+        probe_info_t probeInfo;
+        device_info_t deviceInfo;
+        library_info_t libraryInfo;
+
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_probe_info(probe, &probeInfo), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_device_info(probe, &deviceInfo), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_library_info(probe, &libraryInfo), RTTCouldNotGetDeviceInformation);
+
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.reset(probe, RESET_SYSTEM), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.probe_uninit(&probe), RTTCouldNotGetDeviceInformation);
+        highLevelFunctions.dll_close();
+
+        releaseHighLevel();
+
+        uint32_t clockSpeed = probeInfo.clockspeed_khz;
+        device_family_t family = deviceInfo.device_family;
+        std::string jlinkarmlocation = libraryInfo.file_path;
+
+        RETURN_ERROR_ON_FAIL((nrfjprogdll_err_t)loadnRFjprogFunctions(&dll_function), RTTCouldNotLoadnRFjprogLibrary);
+
+        libraryLoaded = true;
+
+        RETURN_ERROR_ON_FAIL(dll_function.open_dll(jlinkarmlocation.c_str(), &RTT::logCallback, family), RTTCouldNotOpennRFjprogLibrary);
+
+        RETURN_ERROR_ON_FAIL(dll_function.connect_to_emu_with_snr(baton->serialNumber, clockSpeed), RTTCouldNotConnectToDevice);
+        RETURN_ERROR_ON_FAIL(dll_function.connect_to_device(), RTTCouldNotConnectToDevice);
+
+        if (baton->hasControlBlockLocation) {
+            RETURN_ERROR_ON_FAIL(dll_function.rtt_set_control_block_address(baton->controlBlockLocation), RTTCouldNotFindControlBlock);
+        }
+
+        rttStartTime = std::chrono::high_resolution_clock::now();
+        RETURN_ERROR_ON_FAIL(dll_function.rtt_start(), RTTCouldNotStartRTT);
+
+
+        while (true) {
+            bool controlBlockFound = false;
+            RETURN_ERROR_ON_FAIL(dll_function.rtt_is_control_block_found(&controlBlockFound), RTTCouldNotCallFunction);
+            std::chrono::high_resolution_clock::time_point attemptedStartupTime = std::chrono::high_resolution_clock::now();
+
+            if (controlBlockFound) {
+                break;
+            }
+
+            if (std::chrono::duration_cast<std::chrono::seconds>(attemptedStartupTime - rttStartTime).count() > 10) {
+                cleanup();
+                return RTTCouldNotFindControlBlock;
+            }
+        }
+
+        uint32_t downChannelNumber;
+        uint32_t upChannelNumber;
+
+        RETURN_ERROR_ON_FAIL(dll_function.rtt_read_channel_count(&downChannelNumber, &upChannelNumber), RTTCouldNotGetChannelInformation);
+
+        for(uint32_t i = 0; i < downChannelNumber; ++i)
```

Comment:
A `uint32_t` to iterate channel numbers? Are we having more than 65K channels on a RTT? LOL 🤣

Reply:
`uint32_t` covers slightly more then 65K, is it not, or what really is the issue? If the external library uses that type to return channel number, then let it be.

Reply:
This is, as @bencefr also mentions, due to rtt_read_channel_count utilize a `uint32_t`. To avoid the loops producing compile time warnings, it is recommended to use the same type in the loop.